synapse/lib/cache.py

```diff
@@ -10,22 +11,7 @@
 import synapse.common as s_common
 
 def memoize(size=10000):
-
-    def decor(f):
-
-        def callback(args):
-            return f(*args)
-
-        cache = FixedCache(callback, size=size)
-
-        def wrap(*args):
-            return cache.get(args)
-
-        wrap.cache = cache
-
-        return wrap
-
-    return decor
+    return functools.lru_cache(maxsize=size)
```

Comment:
The python lru_cache is quite a bit heavier than our current memoize, including requiring syscall / context switching for every key retrieval.  In many current use cases for this decorator, it would likely get slower.  For situations where the "last recently used" aspect is truly worth a syscall, we should have a different decorator, or just use the one from functools.